int32[6] count
